### PR TITLE
Add iim

### PR DESCRIPTION
**Pull request by monokrome**
##### Motivation for this change

I was trying to install iim and noticed that it wasn't in the package index, so decided that I'd add a package for it.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tomberek** [comment]
@monokrome : cleaned up the derivation, does this meet your intent?

**Comment by monokrome** [comment]
@tomberek Yeah, sorry - it was my first package and I didn't know how to access the things lol. Thank you 🙏🏼

**Comment by monokrome** [comment]
What does the "This PR does not cleanly list package outputs after merging." mean?

**Comment by monokrome** [comment]
Also, this isn't meant to be a draft... I don't know how that happened...

File: pkgs/applications/networking/irc/iim/default.nix
Comment (tomberek) [inline pkgs/applications/networking/irc/iim/default.nix:5]: my mistake, this should be fetchFromGithub (and added to the inputs at the top

**Review by tomberek — CHANGES_REQUESTED** [review]
Fixed up the fetchFromGitHub, this does a more efficient fetching. Do you want to add yourself as a maintainer?

**Comment by monokrome** [comment]
@tomberek Should I add myself as maintainer? Is that the normal thing to do?

**Comment by tomberek** [comment]
> @tomberek Should I add myself as maintainer? Is that the normal thing to do?

Yes. Please make it an independent commit before the current one(same PR is fine).

File: pkgs/applications/networking/irc/iim/default.nix
Comment (monokrome) [reply pkgs/applications/networking/irc/iim/default.nix:5]: @tomberek What does "added to the inputs at the top" mean here? Apologies for my confusion

File: pkgs/applications/networking/irc/iim/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/irc/iim/default.nix:17]: Adding yourself as a maintainer is required.

File: pkgs/applications/networking/irc/iim/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/irc/iim/default.nix:12]:
```suggestion
  installPhase = ''
    install -D iim $out/bin/iim
  '';
```

File: pkgs/applications/networking/irc/iim/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/irc/iim/default.nix:10]: Please use makeFlags.

File: pkgs/applications/networking/irc/iim/default.nix
Comment (SuperSandro2000) [reply pkgs/applications/networking/irc/iim/default.nix:5]: You should use fetchFromGitHub which...

File: pkgs/applications/networking/irc/iim/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/irc/iim/default.nix:2]:
```suggestion
{ lib, stdenv, fetchFromGitHub }:

stdenv.mkDerivation {
```

.. would be added here.

File: pkgs/applications/networking/irc/iim/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/irc/iim/default.nix:3]:
```suggestion
  pname = "iim";
  version ="?";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by monokrome** [comment]
Deleting this because multiple reviewers is annoying and @SuperSandro2000 doesn't know how to write complete sentences

**Comment by SuperSandro2000** [comment]
> Deleting this because multiple reviewers is annoying and @SuperSandro2000 doesn't know how to write complete sentences

It is totally normal to get multiple reviewers if one does not have enough time to take another look at the PR. Normally there are also no contradicting requests/changes which shouldn't make this a problem.

My sentence did start at the one comment and continued over the three dots ... into the other comment. I thought this was clear.

**Pull request closed without merging**
